bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplacer.java

```diff
@@ -0,0 +1,778 @@
+package org.eclipse.ui.texteditor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.PatternSyntaxException;
+
+import org.eclipse.swt.custom.BusyIndicator;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.widgets.Display;
+
+import org.eclipse.jface.text.IFindReplaceTarget;
+import org.eclipse.jface.text.IFindReplaceTargetExtension;
+import org.eclipse.jface.text.IFindReplaceTargetExtension3;
+import org.eclipse.jface.text.IFindReplaceTargetExtension4;
+import org.eclipse.jface.text.IRegion;
+import org.eclipse.jface.text.Region;
+
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.internal.texteditor.NLSUtility;
+
+/**
+ * @since 3.18
+ */
+public class FindReplacer {
+	private FindReplaceStatus status = new FindReplaceStatus();
+	private IFindReplaceTarget fTarget;
+	private IRegion fOldScope;
+	private Point fIncrementalBaseLocation;
+
+	/**
+	 * Wether the search is incremental.
+	 *
+	 * TODO: what is this even doing?!
```

Comment:
you can also call it, search as you type.
if the checkbox is ticked, it immediately searches after each character typed

Reply:
Thank you, this makes sense now! 👍🏼